marlowe-playground-client/webpack.config.js

```diff
@@ -1,145 +1,141 @@
             {
                 test: /\.scss$/,
-                use: ['style-loader', 'css-loader', 'sass-loader']
+                use: [MiniCssExtractPlugin.loader, "css-loader", "sass-loader"],
             },
             {
-                test: /\.(gif|png|jpe?g|svg)$/i,
-                use: 'url-loader'
+                test: /\.(png|svg|jpg|jpeg|gif)$/i,
+                type: "asset/resource",
             },
             {
-                test: /\.ttf$/,
-                use: ['file-loader'],
-            }
-        ]
+                test: /\.(woff|woff2|eot|ttf|otf)$/i,
+                type: "asset/resource",
+            },
+        ],
     },
-
     resolve: {
         modules: [
-            // We need the second entry for node to be able to
-            // locate `node_modules` from client directory when 
-            // modules are referenced from inside `web-common`.
-            'node_modules', path.resolve(__dirname, './node_modules')
+            "node_modules",
+            // We need this entry for node to be able to locate `node_modules` from
+            // client directory when modules are referenced from inside `web-common`.
+            path.resolve(__dirname, "./node_modules"),
         ],
         alias: {
-            grammar: path.resolve(__dirname, './grammar.ne'),
-            static: path.resolve(__dirname, './static'),
-            src: path.resolve(__dirname, './src')
+            grammar: path.resolve(__dirname, "./grammar.ne"),
+            static: path.resolve(__dirname, "./static"),
+            src: path.resolve(__dirname, "./src")
+        },
+        extensions: [".purs", ".js", ".ts", ".tsx"],
+        fallback: {
+            "vm": require.resolve("vm-browserify"),
         },
-        extensions: ['.purs', '.js', '.ts', '.tsx']
     },
-
     resolveLoader: {
         modules: [
-            'node_modules',
-            path.resolve(__dirname, '.')
+            "node_modules",
+            path.resolve(__dirname, ".")
         ]
     },
-
     plugins: [
         new HtmlWebpackPlugin({
-            template: './static/index.html',
-            favicon: 'static/favicon.ico',
-            title: 'Marlowe Playground',
-            productName: 'marlowe-playground',
-            googleAnalyticsId: isWebpackDevServer ? 'UA-XXXXXXXXX-X' : 'UA-119953429-16'
+            template: "./static/index.html",
+            favicon: "static/favicon.ico",
+            title: "Marlowe Playground",
+            productName: "marlowe-playground",
+            googleAnalyticsId: isDevelopment ? "UA-XXXXXXXXX-X" : "UA-119953429-16",
+        }),
+        new MiniCssExtractPlugin({
+            filename: "[name].[contenthash].css",
         }),
         new MonacoWebpackPlugin({
             // note that you have to include typescript if you want javascript to work!
-            languages: ['javascript', 'typescript'],
-        })
-    ].concat(plugins)
+            languages: ["javascript", "typescript"],
```

Comment:
there is no typescript involved (anymore?) is there?

Reply:
JavaScript doesn't work in Monaco without TypeScript (see the comment directly above). In any case the example contracts in the playground are written in TypeScript.